streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -536,6 +537,11 @@ public class StreamsConfig extends AbstractConfig {
     public static final String ROCKS_DB = "rocksDB";
     public static final String IN_MEMORY = "in_memory";
 
+    /** {@code dsl.store.suppliers.class } */
+    public static final String DSL_STORE_SUPPLIERS_CLASS_CONFIG = "dsl.store.suppliers.class";
+    public static final String DSL_STORE_SUPPLIERS_CLASS_DOC = "Defines which store implementations to plug in to DSL operators. Must implement the <code>org.apache.kafka.streams.state.DslStoreSuppliers</code> interface.";
+    public static final Class<?> DSL_STORE_SUPPLIERS_CLASS_DEFAULT = BuiltinDslStoreSuppliers.RocksDbDslStoreSuppliers.class;
```

Comment:
Frankly idk how much it matters, or even if it does at all, but generally we use the class name (ie a `String` for configs whenever possible, eg for defaults. It's just easier to work with strings (and I have a small nagging sense that we may autogenerate docs based on the code here which might struggle with the `Class` variant vs the `String` class name...but I'm not 100% sure)

Reply:
I switched it to string but the ergonomics around it are way worse since I have to do a bunch of manual validation that the config object would do for me. Is there really a reason not to use Class? Even things like the (required) key/value serde configurations are configured as Class:
```
            .define(DEFAULT_KEY_SERDE_CLASS_CONFIG,
                    Type.CLASS,
                    null,
                    Importance.MEDIUM,
                    DEFAULT_KEY_SERDE_CLASS_DOC)
```

Reply:
Ah, sorry, maybe I mislead you here...if it's easier to work with the `Class` directly then go ahead with that